File: oep_spring_backend/src/main/java/com/oep/entities/SystemSettings.java
package com.oep.entities;

import jakarta.persistence.AttributeOverride;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Table(name = "system_settings")
@AttributeOverride(name = "id", column = @Column(name = "settings_id"))
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SystemSettings extends BaseEntity {

    @Column(name = "maintenance_mode", nullable = false)
    private boolean maintenanceMode = false;

    @Column(name = "tab_switch_detection", nullable = false)
    private boolean tabSwitchDetection = true;

    @Column(name = "fullscreen_enforcement", nullable = false)
    private boolean fullscreenEnforcement = true;

    @Column(name = "exam_auto_submit", nullable = false)
    private boolean examAutoSubmit = true;
}

File: AdminServiceDotNET/Models/SystemSettings.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdminServiceDotNET.Models
{
    [Table("system_settings")]
    public class SystemSettings
    {
        [Key]
        [Column("settings_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Column("maintenance_mode")]
        public bool MaintenanceMode { get; set; } = false;

        [Column("tab_switch_detection")]
        public bool TabSwitchDetection { get; set; } = true;

        [Column("fullscreen_enforcement")]
        public bool FullscreenEnforcement { get; set; } = true;

        [Column("exam_auto_submit")]
        public bool ExamAutoSubmit { get; set; } = true;

        [Column("last_updated")]
        public DateTime LastUpdated { get; set; } = DateTime.UtcNow;
    }
}
